import/index_java/src/org/vufind/index/PunctuationContainer.java

```diff
@@ -0,0 +1,78 @@
+package org.vufind.index;
+/**
+ * Singleton for storing punctuation configuration information.
+ *
+ * Copyright (C) Villanova University 2021
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Singleton for storing punctuation configuration information.
+ */
+public class PunctuationContainer
+{
+   private static ThreadLocal<PunctuationContainer> containerCache =
+        new ThreadLocal<PunctuationContainer>()
+        {
+            @Override
+            protected PunctuationContainer initialValue()
+            {
+                return new PunctuationContainer();
+            }
+        };
+
+    private Set<String> punctuationRegEx = new LinkedHashSet<String>();
```

Comment:
If we store patterns here and use `name.replaceAll()` later, the patterns need to be compiled every single time.
For performance reasons it would be better to change the type from `Set<String>` to `Set<java.util.regex.Pattern>` and to call `Pattern.Compile(string)` for each entry only once when populating the set.

Reply:
Good idea. Does 5eb3082  look good to you?

Reply:
Yes that looks better, thanks!